src/backend/src/services/tasks.services.ts

```diff
@@ -147,6 +151,10 @@ export default class TasksService {
 
     if (!isUnderWordCount(notes, 250)) throw new HttpException(400, 'Notes must be less than 250 words');
 
+    if (originalTask.status === 'IN_PROGRESS' && !deadline) {
```

Comment:
I dont think we need this error here. I think we should be checking if the new priority is in progress not the original task.